pkg/cmd/kubeconfig/kubeconfig.go

```diff
@@ -0,0 +1,164 @@
+func NewCmdKubeconfig(f util.Factory, ioStreams util.IOStreams) *cobra.Command {
+	o := newOptions(ioStreams)
+
+	cmd := &cobra.Command{
+		Use:   "kubeconfig",
+		Short: "Print the kubeconfig for the current target",
+		Example: `# Print the kubeconfig for the current target 
+gardenctl kubeconfig
+
+# Print the kubeconfig for the current target in json format
+gardenctl kubeconfig --output json
+
+# Print the Shoot cluster kubeconfig for my-shoot
+gardenctl kubeconfig --garden my-garden --project my-project --shoot my-shoot
+
+# Print the Garden cluster kubeconfig of my-garden. The namespace of the project my-project is set as default
+gardenctl kubeconfig --garden my-garden --project my-project`,
+		RunE: base.WrapRunE(o, f),
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	cmd.Flags().BoolVar(&o.RawByteData, "raw", o.RawByteData, "Display raw byte data")
+	cmd.Flags().BoolVar(&o.Flatten, "flatten", o.Flatten, "Flatten the resulting kubeconfig file into self-contained output (useful for creating portable kubeconfig files)")
+	cmd.Flags().BoolVar(&o.Minify, "minify", o.Minify, "Remove all information not used by current-context from the output")
+	cmd.Flags().StringVar(&o.Context, "context", o.Context, "The name of the kubeconfig context to use")
+
+	utilruntime.Must(cmd.RegisterFlagCompletionFunc("output", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return o.PrintFlags.AllowedFormats(), cobra.ShellCompDirectiveNoFileComp
+	}))
+
+	return cmd
+}
+
+// options is a struct to support kubeconfig command
+type options struct {
+	base.Options
+
+	// CurrentTarget is the current target
+	CurrentTarget target.Target
+
+	// PrintFlags composes common printer flag structs
+	PrintFlags *genericclioptions.PrintFlags
+	// ResourcePrinterFunc is a function that can print objects
+	PrintObject printers.ResourcePrinterFunc
+
+	// Flatten flag the resulting kubeconfig file into self-contained output
+	Flatten bool
+	// Minify flag to remove all information not used by current-context from the output
+	Minify bool
+	// RawByteData flag to display raw byte data
+	RawByteData bool
+
+	// Context holds the name of the kubeconfig context to use
+	Context string
+}
+
+// newOptions returns initialized options
+func newOptions(ioStreams util.IOStreams) *options {
+	return &options{
+		Options: base.Options{
+			IOStreams: ioStreams,
+		},
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme).WithDefaultOutput("yaml"),
+	}
+}
+
+// Complete adapts from the command line args to the data required.
+func (o *options) Complete(f util.Factory, _ *cobra.Command, _ []string) error {
+	manager, err := f.Manager()
+	if err != nil {
+		return err
+	}
+
+	o.CurrentTarget, err = manager.CurrentTarget()
+	if err != nil {
+		return err
+	}
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	o.PrintObject = printer.PrintObj
```

Comment:
Shouldn't we replace our own PrintObject function in the base.Options with this function here. It confuses me that we later call a function that I would assume is implemented by us but it isn't in this case.